cEP-0028.md

```diff
@@ -0,0 +1,479 @@
+        if proxy is not None:
+          return proxy
+
+        try:
+          proxy = FileProxy.from_name(filename, workspace)
+        except OSError as exception:
+          if hard_sync:
+            raise exception
+
+        # possibly raises FileNameError exception
+        proxy = FileProxy(filename, workspace)
+        self.add(proxy)
+
+        return proxy
+    ```
+
+  - `coalib.processes.Processing.get_file_dict()`
+
+    Requires patching `get_file_dict()` to conditionally build
+    file dict from a FileProxyMap instance if available rather
+    than from disk files.
+
+    ```python
+    def get_file_dict(filename_list,
+                      log_printer=None,
+                      allow_raw_files=False,
+                      fileproxy_map=None):
+      """
+      Reads all files into a dictionary.
+
+      :param filename_list:   List of files to get the contents of.
+      :param log_printer:     The logger which logs errors.
+      :param allow_raw_files: Allow the usage of raw files (non text files),
+                              disabled by default
+      :param fileproxy_map:   The FileProxyMap instance to use while building
+                              a file dict when available.
+      :return:                Reads the content of each file into dictionary
+                              with filenames as keys.
+      """
+      file_dict = {}
+      for filename in filename_list:
+        try:
+          if fileproxy_map is not None:
+            proxy = fileproxy_map.resolve(filename,
+                                          hard_sync=True)
+            file_contents = proxy.contents()
+            file_lines = file_contents.splitlines(keepends=True)
+            file_dict[filename] = tuple(file_lines)
+          else:
+            with open(filename, 'r',
+                      encoding=detect_encoding(filename)) as _file:
+              file_dict[filename] = tuple(_file.readlines())
+        except UnicodeDecodeError:
+          if allow_raw_files:
+            file_dict[filename] = None
+            continue
+
+          logging.warning("Failed to read file '{}'. It seems to contain "
+                          'non-unicode characters. Leaving it out.'
+                          .format(filename))
+        except (OSError, FileNameError) as exception:
+          if fileproxy_map is not None and allow_raw_files:
+            file_dict[filename] = None
+            continue
+
+          log_exception("Failed to read file '{}' because of an unknown "
+                        'error. Leaving it out.'.format(filename),
+                        exception,
+                        log_level=LOG_LEVEL.WARNING)
+
+      logging.debug('Files that will be checked:\n' +
+                    '\n'.join(file_dict.keys()))
+
+      return file_dict
+    ```
+
+  - `coalib.misc.Caching`
+
+    Addition of a new `MultiplexedFileCache` class.
+
+    ```python
+    class MultiplexedFileCache(FileCache):
```

Comment:
Since this is tied to your lang-server, maybe name it more specific? `Multiplexed` sounds very general^^